### PR TITLE
Bug Fix: IPv6 Address gets truncated to first 4 characters

### PR DESCRIPTION
**Pull request by mic-max**
Fix Issue #2521

## Changes

NETCORE `ClientIpHeaderTelemetryInitializer` allow parsing of IPv4 and IPv6 addresses from the headers. The resulting IP `requestTelemetry.Context.Location.Ip` will not include a port number.

### Checklist
- [x] I ran Unit Tests locally.
- [x] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Comment by mic-max** [comment]
Should a similar change be made to the WEB version of `ClientIpHeaderTelemetryInitializer` ?

**Comment by TimothyMothra** [comment]
> Should a similar change be made to the WEB version of `ClientIpHeaderTelemetryInitializer` ?

Yes, i think so :)

**Comment by mic-max** [comment]
TODO
- Add failing test cases for non-IP addresses like a URL or other
- Make similar change to WEB version of the same class and unit tests

File: NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/TelemetryInitializers/ClientIpHeaderTelemetryInitializerTests.cs [hunk 1]
```diff
@@ -90,43 +89,36 @@ public void InitializeSetsIpFromRemoteIpAddress()
...
         }
 
-        [Fact]
-        public void InitializeSetsIPFromStandardHeader()
+        [Theory]
+        [InlineData("X-Forwarded-For", "127.0.0.3", null, "127.0.0.3")]
+        [InlineData("X-Forwarded-For", "127.0.0.3:80", null, "127.0.0.3")]
+        [InlineData("X-Forwarded-For", "[::1]:80", null, "::1")]
+        [InlineData("HEADER", "127.0.0.3;127.0.0.4", ",;", "127.0.0.3")]
+        [InlineData("X-Forwarded-For", "bad", null, null)]
```
Comment (TimothyMothra) [inline NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/TelemetryInitializers/ClientIpHeaderTelemetryInitializerTests.cs:97]: Can you include some test cases for IPv6?

Comment (mic-max) [reply NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/TelemetryInitializers/ClientIpHeaderTelemetryInitializerTests.cs:97]: I can add more, `[::1]:80` here was the new ipv6 case I added

Comment (TimothyMothra) [reply NETCORE/test/Microsoft.ApplicationInsights.AspNetCore.Tests/TelemetryInitializers/ClientIpHeaderTelemetryInitializerTests.cs:97]: oh i see. 
sorry, i'm not familiar with ipv6 addresses.

apparently there's two formats:
- short form: `::1`
- long form: `0:0:0:0:0:0:0:1`

i would expect to see unit tests to cover both cases. :)

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs [hunk 2]
```diff
@@ -125,30 +123,17 @@ protected override void OnInitializeTelemetry(HttpContext platformContext, Reque
             telemetry.Context.Location.Ip = requestTelemetry.Context.Location.Ip;
         }
 
-        private static string CutPort(string address)
+        private static bool ParseIPWithPort(string input, out string ipAddressString)
```
Comment (TimothyMothra) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs:126]: nit: for this pattern of a method returning a bool with an out parameter, I would recommend naming it "TryParseIpWithPort".

**Review by TimothyMothra — APPROVED** [review]
LGTM

**Pull request merged**